goal_src/engine/target/target-part.gc

```diff
@@ -5,3 +5,9718 @@
+     :field #x10
+     :flags #x1
+     :random-range #x47800000
+     :random-mult #x3f800000
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x11
+     :flags #x3
+     :initial-value -4
+     :random-mult 1
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x12
+     :flags #x1
+     :initial-value #x42000000
+     :random-range #x42000000
+     :random-mult #x3f800000
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x13
+     :flags #x1
+     :initial-value #x42000000
+     :random-range #x42000000
+     :random-mult #x3f800000
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x14
+     :flags #x1
+     :initial-value #x42000000
+     :random-range #x42000000
+     :random-mult #x3f800000
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x15
+     :flags #x1
+     :random-mult #x3f800000
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x1a
+     :flags #x1
+     :initial-value #x415a740e
+     :random-range #x41da740e
+     :random-mult #x3f800000
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x1c
+     :flags #x1
+     :initial-value #x415a740e
+     :random-range #x415a740e
+     :random-mult #x3f800000
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x1f
+     :flags #x1
+     :initial-value -1039031628
+     :random-range #x4291a2b4
+     :random-mult #x3f800000
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x20
+     :flags #x3
+     :initial-value -4
+     :random-mult 1
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x24
+     :flags #x1
+     :initial-value #x3f5a740e
+     :random-mult #x3f800000
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x2d
+     :flags #x1
+     :initial-value #x3f7ae148
+     :random-mult #x3f800000
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x2e
+     :initial-value #x1c2
+     :random-mult 1
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x2f
+     :initial-value #x4004
+     :random-mult 1
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x32
+     :initial-value 15
+     :random-range 59
+     :random-mult 1
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x33
+     :flags #x6
+     :initial-value #x7d8
+     )
+    (new 'static 'sp-field-init-spec :field #x43)
+    )
+   )
+  )
+
+;; failed to figure out what this is:
+(set!
+  (-> *part-id-table* 2008)
+  (new 'static 'sparticle-launcher
+   :init-specs
+   (new 'static 'inline-array sp-field-init-spec 2
+    (new 'static 'sp-field-init-spec
+     :field #x24
+     :flags #x1
+     :initial-value -1097751884
+     :random-mult #x3f800000
+     )
+    (new 'static 'sp-field-init-spec :field #x43)
+    )
+   )
+  )
+
+;; definition for function process-drawable-burn-effect
+;; Used lq/sq
+(defbehavior process-drawable-burn-effect target ((arg0 int))
+  (sound-play-by-name
+   (the-as
+    sound-name
+    (make-u128 #x6465 (-> (new 'static 'array uint64 1 #x6e7275622d746567) 0))
```

Comment:
this looks wrong.  I would set the label cast as `["uint64", true]`, not as an array.